scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -491,23 +499,20 @@ def get_refs_containing(self, rev: str, pattern: Optional[str] = None):
     def push_refspec(
         self,
         url: str,
-        src: Optional[str],
-        dest: str,
+        refspecs: Union[str, Iterable[str]],
         force: bool = False,
-        on_diverged: Optional[Callable[[str, str], bool]] = None,
         progress: Callable[["GitProgressEvent"], None] = None,
         **kwargs,
-    ):
+    ) -> Mapping[str, int]:
```

Comment:
`SyncStatus` should also be defined in `base.py`, and this should be `Mapping[str, SyncStatus]`